ext/bson/bson_native.c

```diff
@@ -319,6 +687,71 @@ VALUE rb_bson_byte_buffer_get_string(VALUE self)
   return string;
 }
 
+
+VALUE rb_bson_byte_buffer_get_hash(VALUE self){
+  VALUE doc = Qnil;
+  byte_buffer_t *b;
+  char type;
+  VALUE cDocument = rb_const_get(rb_const_get(rb_cObject, rb_intern("BSON")), rb_intern("Document"));
+  TypedData_Get_Struct(self, byte_buffer_t, &rb_byte_buffer_data_type, b);
+
+  /* skip length */
```

Comment:
I'd prefer to see this validate the length -- at confirming (a) the length is less than the remaining length of the buffer and (b) there's a null byte at the end.